controllers/cloud_config_sync_controller_test.go

```diff
@@ -0,0 +1,275 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	infraCloudConfName = "test-config"
+	infraCloudConfKey  = "foo"
+)
+
+func makeInfrastrucutreResource() *configv1.Infrastructure {
+	return &configv1.Infrastructure{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: infrastructureResourceName,
+		},
+		Spec: configv1.InfrastructureSpec{
+			CloudConfig: configv1.ConfigMapFileReference{
+				Name: infraCloudConfName,
+				Key:  infraCloudConfKey,
+			},
+			PlatformSpec: configv1.PlatformSpec{
+				Type: configv1.AzurePlatformType,
+			},
+		},
+	}
+}
+
+func makeInfraCloudConifg() *corev1.ConfigMap {
+	return &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+		Name:      infraCloudConfName,
+		Namespace: openshiftConfigNamespace,
+	}, Data: map[string]string{infraCloudConfKey: "bar"}}
+}
+
+func makeManagedCloudConifg() *corev1.ConfigMap {
+	return &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{
+		Name:      managedCloudConfigMapName,
+		Namespace: openshiftManagedConfigNamespace,
+	}, Data: map[string]string{"cloud.conf": "bar"}}
+}
+
+func TestCloudisCloudConfigEqual(t *testing.T) {
+	reconciler := &CloudConfigReconciler{}
+	assert.True(t, reconciler.isCloudConfigEqual(makeManagedCloudConifg(), makeManagedCloudConifg()))
```

Comment:
Can we use gomega here instead so we don't have a mix of two testing frameworks?